### PR TITLE
Makefile: initial setup 🌮

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

The makefile tries to be *as-reproductible* as possible, aka using specific version of tools, and most importantly, *per-repo* available tools. 

This means it will download and run version of tooling in `$PWD/.bin` without polluting the `GOPATH` or the go modules.

[asciicast](https://asciinema.org/a/315191)

You can override tools too, like if you want to use a specific version of `ko` or a specific version of `golangci-lint` :

```bash
$ make GOLANGCILINT=~/.nix-profile/bin/golangci-lint golangci-lint
# […] 
$ make KO=~/src/github.com/google/ko/ko apply
# […]
```

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Add a `Makefile` to the project
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	$Q $(GO) fmt $(PKGS)
+# Misc
+
+.PHONY: clean
+clean: ; $(info $(M) cleaning…)	@ ## Cleanup everything
+	@rm -rf $(BIN)
+	@rm -rf test/tests.* test/coverage.*
+
+.PHONY: help
+help:
```
Comment (sbwsg) [inline Makefile:105]: What does this one do?  Maybe add a comment?

Comment (vdemeester) [reply Makefile:105]: `make help` to display `help` ? :thinking: :sweat_smile:

Comment (sbwsg) [reply Makefile:105]: There's a lot of regex noise in this entry.  I don't understand what it does.

Comment (vdemeester) [reply Makefile:105]: It's parsing the `Makefile` to display the help :stuck_out_tongue: hence the regex :angel:

**Comment by dlorenc** [comment]
/lgtm

this is awesome!

**Pull request merged**